TeamCode/src/main/java/org/firstinspires/ftc/teamcode/auton/ScoringAlgorithm.java

```diff
@@ -107,25 +110,25 @@ public void resetPos() {
         telemetry.addData("Scoring Status1` ", "Resetting Pos");
 
         robot.<Servo>get("jointServo").setPosition(jointStartPos);
-        robot.<DcMotor>get("armMotor").setDirection(DcMotor.Direction.REVERSE);
-        robot.<DcMotor>get("leftViperSlideMotor").setDirection(DcMotorSimple.Direction.REVERSE);
-        robot.<DcMotor>get("rightViperSlideMotor").setDirection(DcMotorSimple.Direction.REVERSE);
-
-        while (robot.<DcMotor>get("armMotor").getCurrentPosition() != armStartPos) {
-
-            if (robot.<DcMotor>get("rightViperSlideMotor").getCurrentPosition() != viperSlideStartPos) {
-                robot.<DcMotor>get("rightViperSlideMotor").setPower(0.6);
-            } else if (robot.<DcMotor>get("leftViperSlideMotor").getCurrentPosition() == viperSlideStartPos) {
-                robot.<DcMotor>get("leftViperSlideMotor").setPower(0);
-            } else if (robot.<DcMotor>get("leftViperSlideMotor").getCurrentPosition() != viperSlideStartPos) {
-                robot.<DcMotor>get("leftViperSlideMotor").setPower(0.6);
-            } else if (robot.<DcMotor>get("leftViperSlideMotor").getCurrentPosition() == viperSlideStartPos) {
-                robot.<DcMotor>get("leftViperSlideMotor").setPower(0);
-            }
 
+        while (Math.abs(robot.<DcMotor>get("armMotor").getCurrentPosition() - armStartPos) > armStartTolerance) {
+            robot.<DcMotor>get("armMotor").setPower(0.6);
+            telemetry.addData("armMotor Pos", robot.<DcMotor>get("armMotor").getCurrentPosition());
         }
         robot.<DcMotor>get("armMotor").setPower(0);
 
+        while (Math.abs(robot.<DcMotor>get("leftViperSlideMotor").getCurrentPosition() - viperSlideStartPos) > viperSlideStartTolerance) {
+
+            robot.<DcMotor>get("rightViperSlideMotor").setPower(0.4);
```

Comment:
same thing here, right motor is based off of left motor values and cannot run on its own (eg. left motor cannot be stopped while right motor is running)